### ISSUE TITLE
Feature request: Edit recordings / cut list / commercial skip 

### ISSUE THREAD

**Issue by Miwer** [janbar/pvr.mythtv#107]
Back when I was using mythfrontend, an often useful feature was the ability to edit recordings, in order to exclude excess content before or after the recorded show (like when you setup the system to record a couple of minutes before and after the planned program time, to make sure you get it all).

https://www.mythtv.org/wiki/Editing_Recordings

I often used this feature together with the commercial detection (and editing erroneous commercial skips, if any), before transcoding any recordings I wanted to keep, so I would only keep the essential content of the recording, and have the transcoder permanently cut out unneeded content.

I would very much like to see such a feature in the MythTV kodi plugin. I have no idea if this is even possible, or some limitations in Kodi prevents this. But I'd like to think, that at least some crude cutlist editing functionality should be possible.

**Comment by ksooo** [comment]
This should be implemented in Kodi in a way that every add-on can benefit from it, not in the add-on. Every add-on would more or less have to implement the same logic and user interface, which makes not much sense.

**Comment by Miwer** [comment]
@ksooo I totally agree with you. Other PVR plugins could generally benefit from such a feature, if implemented in Kodi. However I think for that to happen, more PVR backends would have to support such a feature. I don't know how many PVRs even support this - of the few I know, I've only seen it in MythTV. Plus different PVRs might implement the functionality different, so there would still be a need for plugin specific coding, and to begin with, some simple interface could be made in the plugin.

But there seems to be another blocker that came to my attention, after creating this feature request.
While MythTV does support this, I think its only available to the native frontend (or other with DB access), since I could not see any API functions available for this. All commflag and cutlist functions are "Get"-only - unless they have some newer functions not yet documented.

https://www.mythtv.org/wiki/API_parameters_29

Tough luck... :(

**Comment by StrvnMrvn** [comment]
I've wanted this functionality since day one of pvr.mythtv but always knew it would require a metric shit ton of work on both sides.

On November 6, 2018 3:59:59 PM UTC, Michael Andersen <notifications@github.com> wrote:
>@ksooo I totally agree with you. Other PVR plugins could generally
>benefit from such a feature, if implemented in Kodi. However I think
>for that to happen, more PVR backends would have to support such a
>feature. I don't know how many PVRs even support this - of the few I
>know, I've only seen it in MythTV. Plus different PVRs might implement
>the functionality different, so there would still be a need for plugin
>specific coding, and to begin with, some simple interface could be made
>in the plugin.
>
>But there seems to be another blocker that came to my attention, after
>creating this feature request.
>While MythTV does support this, I think its only available to the
>native frontend (or other with DB access), since I could not see any
>API functions available for this. All commflag and cutlist functions
>are "Get"-only - unless they have some newer functions not yet
>documented.
>
>https://www.mythtv.org/wiki/API_parameters_29
>
>Tough luck... :(
>
>-- 
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/janbar/pvr.mythtv/issues/107#issuecomment-436305488

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by janbar** [comment]
Yep, the api supply only get for cutlist. It was my contribution for mythtv 0.27, and I didn't imagine kodi would be interested by the feature updating the cut list. It could be possible but I think it would be better to store updated/created cutlist in kodi database instead. The first reason is the life cycle of backend releasing, which is too long and users couldn't have a ready feature even next year. Also setting cutlist can be refused because could be affect the native front.
